keycloak/user-storage-provider/src/main/java/org/eclipse/sw360/keycloak/spi/service/Sw360UserService.java

```diff
@@ -0,0 +1,126 @@
+
+import java.util.List;
+
+import org.apache.thrift.protocol.TCompactProtocol;
+import org.apache.thrift.protocol.TProtocol;
+import org.apache.thrift.transport.THttpClient;
+import org.eclipse.sw360.datahandler.thrift.AddDocumentRequestStatus;
+import org.eclipse.sw360.datahandler.thrift.AddDocumentRequestSummary;
+import org.eclipse.sw360.datahandler.thrift.RequestStatus;
+import org.eclipse.sw360.datahandler.thrift.users.User;
+import org.eclipse.sw360.datahandler.thrift.users.UserGroup;
+import org.eclipse.sw360.datahandler.thrift.users.UserService;
+import org.jboss.logging.Logger;
+
+public class Sw360UserService {
+    private String thriftServerUrl = "http://localhost:8080";
+    private static final Logger logger = Logger.getLogger(Sw360UserService.class);
+
+    public List<User> getAllUsers() {
+        try {
+            UserService.Iface sw360UserClient = getThriftUserClient();
+            return sw360UserClient.getAllUsers();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public User getUserByEmail(String email) {
+        try {
+            UserService.Iface sw360UserClient = getThriftUserClient();
+            return sw360UserClient.getByEmail(email);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public User getUserByEmailOrExternalId(String userIdentifier) {
+        try {
+            UserService.Iface sw360UserClient = getThriftUserClient();
+            return sw360UserClient.getByEmailOrExternalId(userIdentifier, userIdentifier);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public User getUser(String id) {
+        try {
+            UserService.Iface sw360UserClient = getThriftUserClient();
+            return sw360UserClient.getUser(id);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public User getUserByApiToken(String token) {
+        try {
+            UserService.Iface sw360UserClient = getThriftUserClient();
+            return sw360UserClient.getByApiToken(token);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public User getUserFromClientId(String clientId) {
+        try {
+            UserService.Iface sw360UserClient = getThriftUserClient();
+            return sw360UserClient.getByOidcClientId(clientId);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+//    public User getUserFromClientCredentialClientId(String clientId) {
+//        try {
+//            UserService.Iface sw360UserClient = getThriftUserClient();
+//            return sw360UserClient.getByClientId(clientId);
+//        } catch (TException e) {
+//            throw new RuntimeException(e);
+//        }
+//    }
+
+    public User addUser(User user) {
+    	logger.info("Sw360UserService::addUser()::-->"+user);
```

Comment:
We need this debug message? Or maybe move it to `logger.debug`.

Reply:
Done